scripts/gcp_setup/environment/gcloud_env.py

```diff
@@ -212,378 +187,311 @@ def _print_banner(text):
+            print(err)
+            sys.exit(1)
+        else:
             try:
-                orgs = json.loads(out)
-                if orgs:
-                    self._choose_organization(orgs)
-            except ValueError:
-                print('Error retrieving organization id')
+                gcloud_info = json.loads(out)
+                config = gcloud_info.get('config', {})
+                self.project_id = config.get('project')
+                self.authed_user = config.get('account')
+                props = config.get('properties', {})
+                metrics = props.get('metrics', {})
+                self.is_devshell = metrics.get('environment') == 'devshell'
+                print('Got gcloud info')
+            except ValueError as verr:
+                print(verr)
+                sys.exit(1)
+
+    def check_cloudshell(self):
+        """Check whether using Cloud Shell or bypassing Cloud Shell."""
+        if not self.force_no_cloudshell:
+            if not self.is_devshell:
+                print('Forseti highly recommends running this setup within '
+                      'Cloud Shell. If you would like to run the setup '
+                      'outside Cloud Shell, please be sure to do the '
+                      'following:\n\n'
+                      '1) Create a project.\n'
+                      '2) Enable billing for the project.\n'
+                      '3) Install gcloud and authenticate your account using '
+                      '"gcloud auth login".\n'
+                      '4) Set your project using '
+                      '"gcloud config project set <PROJECT_ID>".\n'
+                      '5) Run this setup again, with the --no-cloudshell flag, '
+                      'i.e.\n\n    python setup_forseti.py --no-cloudshell\n')
+                sys.exit(1)
+            else:
+                print('Using Cloud Shell, continuing...')
+        else:
+            print('Bypass Cloud Shell check, continuing...')
+
+    def get_authed_user(self):
+        """Get the current authed user."""
+        if not self.authed_user:
+            print('Error getting authed user. You may need to run '
+                  '"gcloud auth login". Exiting.')
+            sys.exit(1)
+        print('You are: {}'.format(self.authed_user))
+
+    def get_project(self):
+        """Get the project."""
+        if not self.project_id:
+            print('You need to have an active project! Exiting.')
+            sys.exit(1)
+        print('Project id: %s' % self.project_id)
+
+    def get_organization(self):
+        """Infer the organization from the project's parent."""
+        return_code, out, err = self._run_command(
+            ['gcloud', 'projects', 'describe',
+             self.project_id, '--format=json'])
+        if return_code:
+            print(err)
+            print('Error trying to find current organization from '
+                  'project! Exiting.')
+            sys.exit(1)
+
+        try:
```

Comment:
Let's keep the json.loads() inside the try here.  My only question is: if the try fails here, do we want to exit?  Seems if the json.loads() is caught here, it would still be a problem in the following if/elif